kazoo/recipe/lock.py

```diff
@@ -134,7 +134,7 @@ def __init__(self, client, path, identifier=None, extra_lock_patterns=()):
         self._retry = KazooRetry(
             max_tries=None, sleep_func=client.handler.sleep_func
         )
-        self._lock = client.handler.lock_object()
+        self._thread_lock = client.handler.lock_object()
```

Comment:
You asked for a suggestion: how about `_handler_lock`?
Just to recognize that other handlers are not thread based

Reply:
Actually, how about "_acquire_method_lock" which describes what it does instead of what it is.